app/src/main/java/org/thoughtcrime/securesms/util/IasKeyStore.java

```diff
@@ -0,0 +1,33 @@
+package org.thoughtcrime.securesms.util;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import org.thoughtcrime.securesms.push.IasTrustStore;
+import org.whispersystems.signalservice.api.push.TrustStore;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+public final class IasKeyStore {
+
+  private IasKeyStore() {
+  }
+
+  public static KeyStore getIasKeyStore(@NonNull Context context) {
+    try {
+      TrustStore contactTrustStore = new IasTrustStore(context);
+
+      KeyStore keyStore = KeyStore.getInstance("BKS");
+      keyStore.load(contactTrustStore.getKeyStoreInputStream(), contactTrustStore.getKeyStorePassword().toCharArray());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getKeyStoreInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when KeyStore.load throws an exception. Ensure the object returned by *getKeyStoreInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).